aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,360 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/globalaccelerator"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlobalAcceleratorEndpointGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlobalAcceleratorEndpointGroupCreate,
+		Read:   resourceAwsGlobalAcceleratorEndpointGroupRead,
+		Update: resourceAwsGlobalAcceleratorEndpointGroupUpdate,
+		Delete: resourceAwsGlobalAcceleratorEndpointGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"listener_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"endpoint_group_region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"health_check_interval_seconds": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_port": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"health_check_protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					globalaccelerator.HealthCheckProtocolTcp,
+					globalaccelerator.HealthCheckProtocolHttp,
+					globalaccelerator.HealthCheckProtocolHttps,
+				}, false),
+			},
+			"threshold_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"traffic_dial_percentage": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Computed: true,
+			},
+			"endpoint_configurations": {
```

Comment:
Nit: This configuration block attribute name should use a singular form, `endpoint_configuration`.

```suggestion
			"endpoint_configuration": {
```

The difference is configurations currently looking like this:

```hcl
resource "aws_globalaccelerator_endpoint_group" "example" {
  # ... other configuration ...

  endpoint_configurations {
    endpoint_id = "${aws_alb.example1.arn}"
  }

  endpoint_configurations {
    endpoint_id = "${aws_alb.example2.arn}"
  }
}
```

and this:

```hcl
resource "aws_globalaccelerator_endpoint_group" "example" {
  # ... other configuration ...

  endpoint_configuration {
    endpoint_id = "${aws_alb.example1.arn}"
  }

  endpoint_configuration {
    endpoint_id = "${aws_alb.example2.arn}"
  }
}
```